cmd/otel-allocator/internal/collector/collector.go

```diff
@@ -124,6 +127,25 @@ func (k *Watcher) runOnCollectors(store cache.Store, fn func(collectors map[stri
 		if pod.Spec.NodeName == "" {
 			continue
 		}
+
+		isPodUnhealthy := false
```

Comment:
Can we move this check to a helper function? It's a bit more complicated now.

Reply:
Done! Good call out!